### ISSUE TITLE
Any plans to support DTLS 1.2?

### ISSUE THREAD

**Issue by soulofmischief87** [sipwise/rtpengine#186]
I have a call flow that implements slow start causing firefox to take client role and offer DTLS 1.2 after ICE negotiations. According to this post https://bugzilla.mozilla.org/show_bug.cgi?id=1153702 the server side should respond with a server hello and the version of DTLS that it supports. As of now rtpengine does not respond at all to dtls client hello with 1.2 in the version header.

**Comment by rfuchs** [comment]
That's probably just a matter of having support for it in OpenSSL and perhaps enabling it. I'll have to look into it.

**Comment by soulofmischief87** [comment]
Rfuchs,


According to one of the comments in this post. DTLS 1.2 doesn't necessarily have to be supported but the server hello needs to respond with the highest version supported which I think is 1.0 now?

**Comment by rfuchs** [comment]
Yeah and the server hello is generated by OpenSSL. So it needs to be done through OpenSSL somehow.

**Comment by rfuchs** [comment]
From the discussion in that thread, it sounds like it merely is a matter of supporting ECDH, which has been in place for a while now (see #130). Are you perhaps using an older version?

**Comment by rfuchs** [comment]
Sure, because I can't quite make sense of this. If ICE converged on one particular endpoint, then communications should only happen with that endpoint and not do trial and error with several on them.

**Comment by soulofmischief87** [comment]
In this case both ICE peers succeeded and rtpengine is not in the media flow to know wether the other failed or succeeded. Nor was there a reinvite removin I emailed you the traces by the way.
